notes-service/src/main/java/io/meeds/notes/listener/NotesMetadataListener.java

```diff
@@ -0,0 +1,90 @@
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package io.meeds.notes.listener;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.commons.search.index.IndexingService;
+import org.exoplatform.portal.config.UserACL;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.space.spi.SpaceService;
+import org.exoplatform.social.metadata.tag.TagService;
+import org.exoplatform.social.metadata.tag.model.TagName;
+import org.exoplatform.social.metadata.tag.model.TagObject;
+import org.exoplatform.wiki.jpa.search.WikiPageIndexingServiceConnector;
+import org.exoplatform.wiki.model.Page;
+
+import java.util.Set;
+
+public class NotesMetadataListener extends Listener<String, Page> {
+
+  private final IndexingService indexingService;
+
+  private final IdentityManager identityManager;
+
+  private final SpaceService    spaceService;
+
+  private final TagService      tagService;
+
+  private final UserACL         userACL;
+
+  private static final String   NOTES_METADATA_OBJECT_TYPE = "notes";
+
+  public NotesMetadataListener(IndexingService indexingService,
+                               IdentityManager identityManager,
+                               SpaceService spaceService,
+                               TagService tagService,
+                               UserACL userACL) {
+    this.indexingService = indexingService;
+    this.identityManager = identityManager;
+    this.spaceService = spaceService;
+    this.tagService = tagService;
+    this.userACL = userACL;
+  }
+
+  @Override
+  public void onEvent(Event<String, Page> event) {
+    Page note = event.getData();
+    String username = event.getSource();
+    if (username == null) {
+      username = userACL.getSuperUser();
+    }
+    long creatorId = getPosterId(username);
+    long audienceId = getStreamOwnerId(note.getWikiOwner(), username);
+
+    Set<TagName> tagNames = tagService.detectTagNames(note.getContent());
+    tagService.saveTags(new TagObject(NOTES_METADATA_OBJECT_TYPE, note.getId(), note.getParentPageId()),
+                        tagNames,
+                        audienceId,
+                        creatorId);
+    indexingService.reindex(WikiPageIndexingServiceConnector.TYPE, note.getId());
```

Comment:
Please split this listener into two different listeners, each one will consider one Task to do at the same time. Using this combination of Tasks inside a single Listener will lead to not index the content when the Tags throws an error. This is a **best practice** the fact to split into multiple listeners and avoid combining them.

Reply:
you want me to remove the `indexingService.reindex(...` right ? OK i will

Reply:
I have removed the reindex because it happens in the `MetaDataModified` listener even if it doesn't make sense that when the `detectTags` fails(**task1**) we should `reindex`(**task2**).

Reply:
> it doesn't make sense that when the detectTags fails(task1) we should reindex(task2).

Shouldn't we have the content of the updated note searchable when the Tags aren't correct ? Two different Use Cases that shouldn't block each other.